test/sync/src/androidTest/kotlin/io/realm/test/mongodb/shared/SyncSessionTests.kt

```diff
@@ -287,6 +293,95 @@ class SyncSessionTests {
         }
     }
 
+    /**
+     * - Insert a document in MongoDB using the command server
+     * - Fetch the object id of the newly inserted document
+     * - Open a Realm with the same partition key as the inserted document
+     * - Wait for Sync to fetch the document as a valid RealmObject with the matching ObjectId as a PK
+     */
+    @Test
+    fun syncingObjectIdFromMongoDB() {
+        val adminApi = app.asTestApp
+        runBlocking {
+            val config = SyncConfiguration.Builder(user, partitionValue, schema = setOf(ObjectIdPk::class))
+                .build()
+            val realm = Realm.open(config)
+
+            val json: JsonObject = adminApi.insertDocument(
+                ObjectIdPk::class.simpleName!!,
+                """
+                    {
+                        "name": "$partitionValue",
```

Comment:
Again no test against visible validatable data, so this would also pass even though our serialization was off. We need some minimal proof that the round trip yields what we expect.

Reply:
Should be ok after discovering that the `toString`-test is actually verifying that the bytes are serialized correctly ... at least for the hex-string variant.